modules/social_engineering/hta_powershell/module.rb

```diff
@@ -4,25 +4,17 @@
 # See the file 'doc/COPYING' for copying permission
 #
 class Hta_powershell < BeEF::Core::Command
-
   def self.options
+    @config = BeEF::Core::Configuration.instance
+    ps_url = @config.get('beef.extension.social_engineering.powershell.powershell_handler_url')
 
-    @configuration = BeEF::Core::Configuration.instance
-    proto = @configuration.get("beef.http.https.enable") == true ? "https" : "http"
-    beef_host = @configuration.get("beef.http.public") || @configuration.get("beef.http.host")
-    beef_port = @configuration.get("beef.http.public_port") || @configuration.get("beef.http.port")
-    base_host = "#{proto}://#{beef_host}:#{beef_port}"
-
```

Comment:
it doesn't need proto/host/port?

Reply:
It uses `@configuration.beef_url_str` a few lines lower. `@configuration.beef_url_str` is constructed with protocol / host / port.

```ruby
      #
      # Returns the beef scheme://host:port for external resources
      # e.g. hooked browsers
      def beef_url_str
        "#{beef_proto}://#{beef_host}:#{beef_port}"
      end
```